docs/20_smart-contracts/40_guides/10_create_smart_contracts_guide.md

```diff
@@ -0,0 +1,1351 @@
+
+The `hello` smart contract class has only one action implemented by its `hi` public member function.
+
+```c++
+#include <hello.hpp>
+
+ACTION hello::hi( name nm ) {
+   /* fill in action body */
+   print_f("Name : %\n",nm);
+}
+```
+
+The function that implements a smart contract action must be annotated by the `[[eosio::action("action.name")]]` attribute. The `action.name` is optional and if not specified then the action is named by the function name that implements it.
+In the `hello.cpp` generated code the `ACTION` macro is used which expands to `[[eosio::action]] void` C++ code.
+
+The name of the action must:
+
+- Be no longer than 13 characters.
+- Contain only `.`, `a`-`z`, or `1`-`5` characters.
+- May not end with `.`.
+
+Note that when you use the `ACTION` macro the action name is the same as the function name that implements it. Because of that the action name inherits the limitations of the C++ function names as well, which means it can not have `.` in it.
+If you use the `[[eosio::action("action.name")]]` attribute you can name the action differently than the function name that implements it.
+
+## Inline Actions
+
+An inline action is an action that is executed within the context of a parent action. This means that an inline action is initiated by a smart contract action and is executed within the same transaction as the parent action.
+
+Inline actions are useful in situations where a smart contract action needs to interact with another smart contract. Instead of making an external call to the other contract, which could potentially result in a new transaction, the action can be executed inline within the same transaction.
+
+If any part of the transaction fails, the inline action will unwind with the rest of the transaction. Also because they run in the context of the current transaction the order of execution of the inline actions it is guaranteed.
+
+The easiest way to execute an inline action is to use `SEND_INLINE_ACTION` macro.
+
+Let's extend the hello smart contract to:
+
+- Implement a new action called `gettime` which has no input parameter and prints the current date and time at the console.
+- Modify the `hi` action to send an inline `gettime` action to the blockchain.
+
+```hpp
+#include <eosio/eosio.hpp>
+using namespace eosio;
+
+CONTRACT hello : public contract {
+   public:
+      using contract::contract;
+
+      ACTION hi(name nm);
+      ACTION gettime();
+
+   private:
+      void parse_time(uint64_t epoch_time, uint16_t& year, uint8_t& month, uint8_t& day, uint8_t& hour, uint8_t& minute, uint8_t& second);
+};
+```
+
+```cpp
+#include <hello.hpp>
+#include <eosio/eosio.hpp>
+#include <eosio/system.hpp>
+#include <eosio/time.hpp>
+
+ACTION hello::hi(name nm) {
+
+   print_f("Name : %\n", nm);
+
+   SEND_INLINE_ACTION(*this, gettime, {get_self(), "active"_n}, {});
+}
+
+ACTION hello::gettime() {
```

Comment:
This **really** needs to be simplified. 

There's no reason to have such a verbose function here for the purpose of explaining inline actions. It could just be another printf action which would give the dev the same outcome (2 logs), but would significantly reduce the length of this guide.

Reply:
changed to a printf